src/python/pants/build_graph/meta_rename.py

```diff
@@ -0,0 +1,69 @@
+# coding=utf-8
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from collections import defaultdict
+
+from pants.backend.jvm.targets.scala_library import ScalaLibrary
+from pants.base.specs import DescendantAddresses
+from pants.build_graph.address import Address
+from pants.task.task import Task
+
+class MetaRename(Task):
+  """Rename a target for its dependants
```

Comment:
Given the philosophy that Pants generally does not modify files, (with the exception of `fmt` goal, but it does not modify BUILD files), it would be good to warn users not to mix this goal with other operations.

`Rename a target for its dependants. WARNING: do not use with other goals since BUILD file may be modified`